File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int minVal = INT_MAX;
        for (int i : prices) {
            profit = max(profit, i - minVal);
            minVal = min(minVal, i);
        }
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ans = 0
        low = prices[0]
        r = 0
        while r < len(prices):
            ans = max(prices[r] - low, ans)
            if prices[r] < low:
                low = prices[r]
            r = r + 1
        return ans